libs/blocks/notification/notification.js

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/*
+* Notification - v1.0
+*/
+
+import { decorateBlockText, decorateBlockBg, decorateTextOverrides } from '../../utils/decorate.js';
+import { createTag } from '../../utils/utils.js';
+
+const variants = ['banner', 'ribbon', 'pill'];
+const sizes = ['small', 'medium', 'large'];
+const [banner, ribbon, pill] = variants;
+const [small, medium, large] = sizes;
+const defaultSize = medium;
+const defaultVariant = banner;
+const blockConfig = {
+  [banner]: {
+    [small]: ['s', 's', 's', 'm'],
```

Comment:
Could you share some context on why the small banner could be `s` or `m`? I don't understand the need for the mapping

Reply:
The mapping like this is mostly due to the difference in button sizes (the final value in the array) across each size variant. If it weren't for that, we could just assume that a `small` notification had `s` for all four values, but there's not such thing as a small button in styles.css. But either way, `decorateBlockText` expects an array for a config, and the default (`['m', 's', 'm']`) doesn't work for this block.